File: articles/calculus-for-systems-modeling/articles/carbon-accumulation-and-emissions-pathways/cpp/carbon_accumulation_pathways.cpp
#include <algorithm>
#include <cmath>
#include <iostream>

double linear_decline(double e0, int year, int years){
  return std::max(0.0, e0 * (1.0 - static_cast<double>(year) / static_cast<double>(years)));
}

int main(){
  double e0 = 40.0;
  int years = 30;
  double cumulative = 0.0;
  for(int y=0; y<=years; ++y) cumulative += linear_decline(e0, y, years);
  std::cout << "scenario_name,pathway_type,cumulative_emissions,warning\n";
  std::cout << "linear_decline_to_zero,linear_decline," << cumulative << ",linear_decline_still_accumulates_until_net_zero\n";
}

File: articles/calculus-for-systems-modeling/articles/carbon-accumulation-and-emissions-pathways/fortran/carbon_accumulation_pathways.f90
program carbon_accumulation_pathways
  implicit none
  integer :: y, years
  real(8) :: e0, cumulative, emission
  e0 = 40.0d0
  years = 30
  cumulative = 0.0d0
  do y = 0, years
    emission = max(0.0d0, e0 * (1.0d0 - dble(y) / dble(years)))
    cumulative = cumulative + emission
  end do
  print '(A)', 'scenario_name pathway_type cumulative_emissions warning'
  print '(A,1X,A,1X,F12.6,1X,A)', 'linear_decline_to_zero','linear_decline',cumulative,'linear_decline_still_accumulates_until_net_zero'
end program
